monai/bundle/config_reader.py

```diff
@@ -0,0 +1,157 @@
+    def load_config_file(cls, filepath: PathLike, **kwargs):
+        """
+        Load config file with specified file path (currently support JSON and YAML files).
+
+        Args:
+            filepath: path of target file to load, supported postfixes: `.json`, `.yml`, `.yaml`.
+            kwargs: other arguments for ``json.load`` or ```yaml.safe_load``, depends on the file format.
+
+        """
+        _filepath: str = str(Path(filepath))
+        if not cls.path_match.findall(_filepath):
+            raise ValueError(f'unknown file input: "{filepath}"')
+        with open(_filepath) as f:
+            if _filepath.lower().endswith(cls.suffixes[0]):
+                return json.load(f, **kwargs)
+            if _filepath.lower().endswith(cls.suffixes[1:]):
+                return yaml.safe_load(f, **kwargs)
+            raise ValueError(f"only support JSON or YAML config file so far, got name {_filepath}.")
+
+    @classmethod
+    def load_config_files(cls, files: Union[PathLike, Sequence[PathLike], dict], **kwargs) -> dict:
+        """
+        Load config files into a single config dict.
+
+        Args:
+            files: path of target files to load, supported postfixes: `.json`, `.yml`, `.yaml`.
+            kwargs: other arguments for ``json.load`` or ```yaml.safe_load``, depends on the file format.
+        """
+        if isinstance(files, dict):  # already a config dict
+            return files
+        content = {}
+        for i in ensure_tuple(files):
+            content.update(cls.load_config_file(i, **kwargs))
+        return content
+
+    @classmethod
+    def export_config_file(cls, config: Dict, filepath: PathLike, fmt="json", **kwargs):
+        """
+        Export the config content to the specified file path (currently support JSON and YAML files).
+
+        Args:
+            config: source config content to export.
+            filepath: target file path to save.
+            fmt: format of config content, currently support ``"json"`` and ``"yaml"``.
+            kwargs: other arguments for ``json.dump`` or ``yaml.safe_dump``, depends on the file format.
+
+        """
+        _filepath: str = str(Path(filepath))
+        writer = look_up_option(fmt.lower(), {"json", "yaml"})
+        with open(_filepath, "w") as f:
+            if writer == "json":
+                return json.dump(config, f, **kwargs)
+            if writer == "yaml":
+                return yaml.safe_dump(config, f, **kwargs)
+            raise ValueError(f"only support JSON or YAML config file so far, got {writer}.")
+
+    @classmethod
+    def split_path_id(cls, src: str) -> Tuple[str, str]:
+        """
+        Split `src` string into two parts: a config file path and component id.
+        The file path should end with `(json|yaml|yml)`. The component id should be separated by `#` if it exists.
+
+        Args:
+            src: source string to split.
+
+        """
+        path, *ids = f"{src}".rsplit(ReferenceResolver.sep, 1)
```

Comment:
Hi @wyli ,

I feel this logic may not be correct, the `src` can be something like:
`"~/data/nic#work/config.json#net#<args>#channels"`
Maybe we should use `cls.path_match.findall` directly on `src`?

Thanks.

Reply:
Sure, I feel we need a refactoring for this to unify the ref/macro/I'd syntax, could you work on this?

Reply:
I updated this function logic in the latest commit.

Thanks.